### PR TITLE
Add support for Two-Factor-Authentication

### LINKED ISSUE #86: Support for two-step authentication?
Seems like QuickRadar does not support two-step authentication? Or is something wrong with my setup? I'm getting Authentication Error.

### PR DESCRIPTION
**Pull request by steipete**
This PR adds support for two-factor authentication. This should close #86, even if the ticket mentions two-step-auth; which is super confusing a different, older 2-* auth variant from Apple.

Most people will use the modern one that pops up an alert.

This now works with my account, but should be tested with a few more people.

This uses API that's 10.9 and higher, I assume that's fine? AppKit is already dirty enough :)

Binary (DEBUG release, but performance doesn't really matter here)
https://cl.ly/102644223v3y

*Please help test this and report if it works in the comments*

### CODE REVIEW

**Comment by dlo** [comment]
Works for me!

**Comment by simonrice** [comment]
Just tested and it works for me, and this is the first time I've just submitted a radar with this tool 🎉

**Comment by chockenberry** [comment]
One minor nit: the button in the dialog that takes the 2FA code should be labeled "OK", not "Ok".

**Comment by chockenberry** [comment]
Also, it appears that the 2FA code isn't being persisted. Each time I submit a Radar, I get the 2FA map and code.

This may be a general problem with RadarWeb right now, because I get the 2FA map and code each time I login to the web, too.

**Comment by steipete** [comment]
Force-pushed to fix the "Ok" bug. *shame bell*

**Comment by steipete** [comment]
I never managed to get the browser to be trusted for Radar. Filed [rdar://29131649](http://openradar.appspot.com/29131649) a while back on that. Feel free to dupe - would be so great if this could persist for 30 days.

**Comment by dlo** [comment]
Personally I think the real issue is that you're prompted at all on the machine you're trying to log in with. I've always wondered, doesn't that defeat the entire purpose of 2FA?

**Comment by chockenberry** [comment]
@steipete Yep, I'll definitely dupe that (and it's Exhibit #15432 that Apple doesn't really get web services.)

**Comment by steipete** [comment]
@dlo You're prompted on up to 5 "trusted" devices, configurable.
@chockenberry `digit1` `digit2` `digit3` `digit4` `digit5` `digit6` BECAUSE 😂

**Comment by chockenberry** [comment]
@steipete Every time Apple releases a new web product, they have 2FA growing pains. Search Ads and RadarWeb are just the latest.

Compare that with Google that has ONE screen for doing 2FA for ALL their various products.

Which do you think is easier to keep secure and up-to-date?

**Comment by steipete** [comment]
Also relevant: The [login-variant](https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/two_step_client.rb) that @KrauseFx used in fastlane is quite different (but again, different part of Apple's backend)

**Comment by steipete** [comment]
@amyworrall Did you had a chance to look into that yet?

File: QuickRadar/QRRadarSubmissionService.m [hunk 1]
```diff
@@ -194,7 +196,81 @@ - (void)submitAsyncWithProgressBlock:(void (^)())progressBlock completionBlock:(
 			});
 		}
         
-        
+        // ------- 2-Factor-Auth Support by @steipete --------
+
+        if ([bouncePage.pageContent containsString:@"protected with two-factor authentication"]) {
```
Comment (amyworrall) [inline QuickRadar/QRRadarSubmissionService.m]: We'll have to keep an eye on that one if they change the strings. But I guess that's true for all of this scraping :)

**Comment by steipete** [comment]
Thanks for merging! I think there's a new issue with the componentID missing, but that's unrelated to this one - will open a new ticket once I find out more. The fun never stops :)

**Pull request merged**